### DISCUSSION TITLE
Component does not re-render / state gets updated...

### DISCUSSION THREAD

**Discussion by fastsyrup** [pmndrs/zustand#1616] [Q&A]
Hi there, 
Don't know what I'm doing wrong, the data in my zustand store gets updated, but the component doesn't re-render.

Here's my code:

useNoteBookStore.tsx:
```
export const useNotebookStore = (notebookUID: uid | undefined) => {
  if (!notebookUID) return undefined;
  const name = "nb-" + notebookUID;
  const store = create<NotebookStoreType>()(
    devtools(
      persist(
        immer((set) => ({
          data: welcomeData,
          resetNotebook: (data) => set({ data }),
          setNodeText: (nodeUID, text) => {
            set((state) => {
              const nodeIndex = state.data.content.findIndex(
                (item) => item.uid === nodeUID
              );
              state.data.content[nodeIndex].text = text;
            });
          },
        })),
        { name: name }
      )
    )
  );
  return store;
};
```

Here's how I use it in a <EditorView> functional component:
```
...
const store = useNotebookStore(pane?.uuid);
const data = store ? store((state) => state.data) : undefined;

if (!pane || !pane.uuid || !data) return <DefaultView />;

const nodeList = data.content.map((item) => {
    return <TextNode item={item} notebookUID={pane.uuid} />;
 });
...
Render the array of TextNode (nodeList) in another component...
```
Now when I call setNodeText(..) it will update the node in the data of the store ( I see it in devtools) but the components  will not rerender.

Info: Since I'm storing many notebooks, I need to pass the NoteBook UID to the store in order to load / save only the one that is active.  That's why I wrote this simple wrapper which dynamically calculates the right key (name) where to persist the store in localstorage.

What am I doing wrong?

**Comment by dai-shi** [comment]
You are creating a new fresh store every time it renders. Consider creating store outside hooks / components.

**Reply by fastsyrup** [reply]
Thanks @dai-shi, I wasn't aware of that.
